DevTools/HackStudio/Editor.cpp

```diff
@@ -480,8 +481,17 @@ void Editor::set_document(GUI::TextDocument& doc)
         set_syntax_highlighter(nullptr);
     }
 
-    if (m_language_client)
-        m_language_client->open_file(code_document.file_path().string());
+    if (m_language_client) {
+        auto full_file_path = String::formatted("{}/{}", project().root_directory(), code_document.file_path());
+        dbg() << "Opening " << full_file_path;
+        int fd = open(full_file_path.characters(), O_RDONLY | O_NOCTTY);
```

Comment:
do we actually set the controlling terminal to a terminal device when it's opened? I didn't expect to see `O_NOCTTY` here...?
for that matter, why is the user opening a non-regular file with hackstudio (and why are we allowing this)?

Reply:
Right, we don't seem to actually check it anywhere. We do have `Process::tty`, but that doesn't seem to influence anything either?

Perhaps let's just unimplement that and ignore controlling terminals the same way we ignore locales?

> for that matter, why is the user opening a non-regular file with hackstudio (and why are we allowing this)?

Perhaps some other code should do the check and disallow this (or does it already?), but the user can put a path to any kind of file into the project file...

Reply:
I think I'd keep this specific `open()` behaviour out, at least.

Some other code _should_ check that and ensure that the user isn't trying to edit `/dev/urandom`, yeah :P